src/type-system.ts

```diff
@@ -354,6 +361,33 @@ export const ElysiaType = {
 			})
 			.Encode((value) => value) as any as TNumber
 	},
+	Integer: (property?: NumberOptions): TInteger => {
+		const schema = Type.Integer(property)
+
+		return t
+			.Transform(
+				t.Union(
+					[
+						t.String({
+							format: 'integer',
+							default: 0
+						}),
+						t.Number(property)
```

Comment:
probably a typo? should be `Integer` and not `Number`

Reply:
adding `t.Integer` causes infinite recursion:

```ts
✗ Params Validator > parse malformed integer [3.00ms]
370 |                                   [
371 |                                           t.String({
372 |                                                   format: 'integer',
373 |                                                   default: property?.default ?? 0
374 |                                           }),
375 |                                           t.Integer(property)
              ^
RangeError: Maximum call stack size exceeded.
```

Reply:
yeah using `t.Integer` will lead to infinite recursion, but you could do `Type.Integer`.

Reply:
oops, i've just seen it, noted.